### PR TITLE
MAINT: reduce code duplication; make common skip func

### PR DESCRIPTION
**Pull request by samir-nasibli**
## Description

Follow up for #2076. See https://github.com/intel/scikit-learn-intelex/pull/2076#pullrequestreview-2347708109

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

**Review by Alexsandruss — APPROVED** [review]

File: sklearnex/tests/test_run_to_run_stability.py [hunk 1]
```diff
@@ -146,6 +146,18 @@ def _run_test(estimator, method, datasets):
...
+        not _IS_INTEL
+        and ("Neighbors" in estimator or "LocalOutlierFactor" in estimator)
+        and method in ["score", "predict", "kneighbors", "kneighbors_graph"]
+    ):
+        if daal_check_version((2025, "P", 200)):
+            pytest.fail("Re-verify failure of algorithms in oneDAL 2025.2")
+        pytest.skip(f"{estimator} shows instability on non-Intel(R) hardware")
+
+
+
```
Comment (samir-nasibli) [inline sklearnex/tests/test_run_to_run_stability.py:160]:
```suggestion
```
